### PR TITLE
Replace max_shrinks=0 with Phase setting in our tests

### PR DESCRIPTION
**Pull request by Zac-HD**
This pull is step one of a two-stage plan to deal with the `max_shrinks` setting (see #1235):

1. Do the easy part, by substantially reducing the use of `max_shrinks` in our own test suite.
2. Leave the hard parts for ~~someone else~~ later!  At least it will be a smaller job after this.

### CODE REVIEW

File: hypothesis-python/tests/common/debug.py [hunk 1]
```diff
@@ -76,7 +77,7 @@ def find_any(
     settings = Settings(
         settings,
         max_examples=10000,
-        max_shrinks=0000,
+        phases=tuple(Phase)[:Phase.shrink],
```
Comment (DRMacIver) [inline hypothesis-python/tests/common/debug.py]: I'm not a huge fan of that way of writing this. It's pretty fragile against adding new phases. Why not `set(Phase) - {Phase.shrink}`?

File: hypothesis-python/tests/common/utils.py [hunk 2]
```diff
@@ -23,10 +23,13 @@
 from io import BytesIO, StringIO
 
 from hypothesis.errors import HypothesisDeprecationWarning
+from hypothesis._settings import Phase
 from hypothesis.reporting import default, with_reporter
 from hypothesis.internal.compat import PY2
 from hypothesis.internal.reflection import proxies
 
+no_shrink = tuple(Phase)[:Phase.shrink]
```
Comment (DRMacIver) [inline hypothesis-python/tests/common/utils.py]: Why are there the other `tuple(Phase)` values rather than just importing this?

**Comment by Zac-HD** [comment]
I've fixed some import-related issues caused by extracting a subset of a larger commit, and also made the definition of `no_shrink` more robust as suggested.

**Comment by Zac-HD** [comment]
:tada:

**Pull request merged**